tests/test_tensornet_tf.py

```diff
@@ -0,0 +1,579 @@
+
+        obs = getattr(qml.ops, name)
+
+        assert tensornet_tf_device_1_wire.supports_observable(name)
+
+        @qml.qnode(tensornet_tf_device_1_wire)
+        def circuit():
+            qml.QubitStateVector(np.array(state), wires=[0])
+            return qml.expval(obs(*par, wires=[0]))
+
+        assert np.isclose(circuit(), expected_output, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("name,state,expected_output,par", [
+        ("Hermitian", [1 / math.sqrt(3), 0, 1 / math.sqrt(3), 1 / math.sqrt(3)], 5 / 3,
+         [np.array([[1, 1j, 0, 1], [-1j, 1, 0, 0], [0, 0, 1, -1j], [1, 0, 1j, 1]])]),
+        ("Hermitian", [0, 0, 0, 1], 0, [np.array([[0, 1j, 0, 0], [-1j, 0, 0, 0], [0, 0, 0, -1j], [0, 0, 1j, 0]])]),
+        ("Hermitian", [1 / math.sqrt(2), 0, -1 / math.sqrt(2), 0], 1,
+         [np.array([[1, 1j, 0, 0], [-1j, 1, 0, 0], [0, 0, 1, -1j], [0, 0, 1j, 1]])]),
+        ("Hermitian", [1 / math.sqrt(3), -1 / math.sqrt(3), 1 / math.sqrt(6), 1 / math.sqrt(6)], 1,
+         [np.array([[1, 1j, 0, .5j], [-1j, 1, 0, 0], [0, 0, 1, -1j], [-.5j, 0, 1j, 1]])]),
+        ("Hermitian", [1 / math.sqrt(2), 0, 0, 1 / math.sqrt(2)], 1,
+         [np.array([[1, 0, 0, 0], [0, -1, 0, 0], [0, 0, -1, 0], [0, 0, 0, 1]])]),
+        ("Hermitian", [0, 1 / math.sqrt(2), -1 / math.sqrt(2), 0], -1,
+         [np.array([[1, 0, 0, 0], [0, -1, 0, 0], [0, 0, -1, 0], [0, 0, 0, 1]])]),
+    ])
+    def test_supported_observable_two_wires_with_parameters(self, tensornet_tf_device_2_wires, tol, name, state,
+                                                            expected_output, par):
+        """Tests supported observables on two wires with parameters."""
+
+        obs = getattr(qml.ops, name)
+
+        assert tensornet_tf_device_2_wires.supports_observable(name)
+
+        @qml.qnode(tensornet_tf_device_2_wires)
+        def circuit():
+            qml.QubitStateVector(np.array(state), wires=[0, 1])
+            return qml.expval(obs(*par, wires=[0, 1]))
+
+        assert np.isclose(circuit(), expected_output, atol=tol, rtol=0)
+
+    def test_cannot_overwrite_state(self, tensornet_tf_device_2_wires):
+        """Tests that _state is a property and cannot be overwritten."""
+
+        dev = tensornet_tf_device_2_wires
+
+        with pytest.raises(AttributeError, match="can't set attribute"):
+            dev._state = np.array([[1, 0],
+                                   [0, 0]])
+
+    def test_correct_state(self, tensornet_tf_device_2_wires):
+
+        dev = tensornet_tf_device_2_wires
+        state = dev._state
+
+        expected = np.array([[1, 0],
+                             [0, 0]])
+        assert np.allclose(state, expected)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.Hadamard(wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        circuit()
+        state = dev._state
+
+        expected = np.array([[1, 0],
+                             [1, 0]]) / np.sqrt(2)
+        assert np.allclose(state, expected)
```

Comment:
Since this almost completely duplicates tests from `test_tensornet.py` (which itself duplicates `test_default_qubit.py`), it would be good to make these tests uniform (and ideally extend to external plugins)